### ISSUE TITLE
Help with CT Paired lung

### ISSUE THREAD

**Issue by MichaelWerthmann** [DeepRegNet/DeepReg#764]
## Subject of the issue
Hello,

I am relatively new to Deep Learning. For a student project I want to register some paired lung CT volumes without training my own model. Therefore, I wanted to use the pre trained network from your demo. I preprocessed my data the same way as described in the load_data python script and put them with according labels in the test directory. I downloaded the pre-trained model and then I started the predict script and got some results. However the result image looks nearly identical to the moving image. I don't know if the registration is just that good or maybe something went wrong. Also I got some warnings/errors about "two checkpoint references resolved to different objects".

Is this approach ok or is it totally wrong to use the pretrained model? Is there any possibility how I can achieve DeepReg Results of my lung CTs more easily? 

I am thankful for every help.

Kind Regards
Michael

**Comment by YipengHu** [comment]
Hi @MichaelWerthmann 

First, the model you downloaded is a demo, with limited training set and unverified generalisation ability to other data, say from a different scanner. Maybe first compare your pre-processed data with the data in the demo see how much difference are there, field-of-view, orientation, CT window, to name a few.

Second, when you say "nearly identical", do you think it is much smaller than the difference between moving and warped images in the demo test? I'm not sure about the warning, which needs further investigation.

In your case, perhaps a more traditional iterative method could be a starting point to see if these can be registered - DeepReg has most functions for this as well, for example: https://deepreg.readthedocs.io/en/latest/demo/classical_mr_prostate_nonrigid.html. If you do have a substantial set of data for training, I would be interested to see how DeepReg could work on them using a learning-based method.

**Comment by MichaelWerthmann** [comment]
Hello @YipengHu,

thank you very much for your quick answer. I'll give you more background of my project. My goal is to compare multiple lung registration algorithms. Therefore, I think it would be nice to also include a Deep Learning based algorithm. I have about 40 scans but and they have quite different sizes and spacings and sometimes different FOVs. Orientation wise they are the same. 

I do not want to make it too complicated because DL is not the essential part of my work, so I thought that maybe I could just grab some pre-trained model and apply it. But unverified generalisation does not sound too good for my plans. Do you know of an easy to apply pre trained lung model for registration?  

Regarding the classical method. What do you think would be the best way to apply it to a folder of niftis? 

Kind Regards
Michael

**Comment by YipengHu** [comment]
@MichaelWerthmann 
- i do not have a pre-trained model which is robust enough to be applied to data from other sources. I don't know there is one out there unfortunately. But with 40 (pairs?) you might have a chance to train your own model, with or without these publicly available data sets. Happy to discuss further and support your research if this is something you are interested in.
- for the classical method, you need to follow the script which does not support a software-style command line tool (for now). I'm sure you are aware there are many other registration tools for pairwise iterative registration algorithms.
